File: src/main/java/io/github/takenoha/towerdefense/paper/RaidSealMaterialPolicy.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;

/** Material compatibility policy for the current and legacy raid-seal items. */
public final class RaidSealMaterialPolicy {
    public static final String CURRENT_MATERIAL = "ECHO_SHARD";
    public static final String LEGACY_MATERIAL = "ENDER_EYE";

    private RaidSealMaterialPolicy() {
    }

    public static boolean supports(String materialName) {
        Objects.requireNonNull(materialName, "materialName");
        return CURRENT_MATERIAL.equals(materialName) || LEGACY_MATERIAL.equals(materialName);
    }

    public static boolean isLegacy(String materialName) {
        Objects.requireNonNull(materialName, "materialName");
        return LEGACY_MATERIAL.equals(materialName);
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/RaidSealMaterialPolicy.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects

/** Material compatibility policy for the current and legacy raid-seal items. */
class RaidSealMaterialPolicy private constructor() {
    companion object {
        const val CURRENT_MATERIAL: String = "ECHO_SHARD"
        const val LEGACY_MATERIAL: String = "ENDER_EYE"

        @JvmStatic
        fun supports(materialName: String): Boolean {
            Objects.requireNonNull(materialName, "materialName")
            return CURRENT_MATERIAL == materialName || LEGACY_MATERIAL == materialName
        }

        @JvmStatic
        fun isLegacy(materialName: String): Boolean {
            Objects.requireNonNull(materialName, "materialName")
            return LEGACY_MATERIAL == materialName
        }
    }
}
